### PR TITLE
feat: Twitch OAuth with lucia

### PR DESCRIPTION
**Pull request by okdargy**
example of .env if you want to replicate yourself:
```
TWTICH_CLIENT_ID=""
TWITCH_CLIENT_SECRET=""
DATABASE_URL="mysql://root:dargy@localhost:3306/greasy"
```

### CODE REVIEW

File: pages/index.vue [hunk 1]
```diff
@@ -13,6 +13,10 @@
       icon: "#i-spotify",
       href: "https://open.spotify.com/playlist/21phkh5dZrZtO4E9Bf9qUy",
     },
+    {
+      icon: "#i-lock",
+      href: "/dashboard",
+    }
```
Comment (veryCrunchy) [inline pages/index.vue]:
```suggestion
    }
```
Dashboard should not be linked to; viewers will be seeing this page.

File: pages/index.vue [hunk 2]
```diff
@@ -60,6 +64,11 @@
...
       />
     </symbol>
+    <symbol id="i-lock" viewBox="0 0 448 512">
+      <path
+        d="M144 144v48H304V144c0-44.2-35.8-80-80-80s-80 35.8-80 80zM80 192V144C80 64.5 144.5 0 224 0s144 64.5 144 144v48h16c35.3 0 64 28.7 64 64V448c0 35.3-28.7 64-64 64H64c-35.3 0-64-28.7-64-64V256c0-35.3 28.7-64 64-64H80z"
+      />
+    </symbol>
```
Comment (veryCrunchy) [inline pages/index.vue]:
```suggestion

```

File: prisma/schema.prisma
Comment (veryCrunchy) [inline prisma/schema.prisma:1]: We should consider using drizzle instead as prisma is quite slow

File: server/routes/login/twitch.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { generateState } from 'arctic';
+import { twitch } from '~/server/utils/auth';
+
+export default defineEventHandler(async (event) => {
+    const state = generateState();
+    const url = await twitch.createAuthorizationURL(state, {
+        scopes: ['channel:read:subscriptions', 'channel:read:subscriptions']
```
Comment (veryCrunchy) [inline server/routes/login/twitch.ts]: Did you intend to put duplicate scopes?

**Review by veryCrunchy — CHANGES_REQUESTED** [review]

File: server/routes/login/twitch.ts (hunk 3 above)
Comment (okdargy) [reply server/routes/login/twitch.ts]: Nope, my bad.

File: pages/index.vue (hunk 1 above)
Comment (okdargy) [reply pages/index.vue]: It's more of a temporary thing because I'm assuming the whole frontpage will be redesigned

File: prisma/schema.prisma
Comment (okdargy) [reply prisma/schema.prisma:1]: Alright, let me cook...

File: pages/index.vue (hunk 1 above)
Comment (veryCrunchy) [reply pages/index.vue]: Eventually, will be updating the site to this very soon tho.

**Comment by veryCrunchy** [comment]
I believe in you

**Comment by okdargy** [comment]
Only thing needed now is to lint all the files to pass all the tests. Please review a4e5d6e, boss. Thanks.

File: server/utils/auth.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import { sessionTable, userTable } from "./schema";
+import { drizzle } from "drizzle-orm/mysql2";
+import mysql from "mysql2";
+
+const connection = mysql.createConnection({
+	host: process.env.DB_HOST || "localhost",
+	user: process.env.DB_USER || "root",
+	password: process.env.DB_PASS || "",
+	database: process.env.DB_NAME || "greasy",
+});
```
Comment (veryCrunchy) [inline server/utils/auth.ts:15]: We should create the db connection in a dedicated file so we can access it everywhere

**Review by veryCrunchy — APPROVED** [review]

**Pull request merged**